mlrun/projects/project.py

```diff
@@ -3446,6 +3448,85 @@ def remove_custom_packager(self, packager: str):
         """
         self.spec.remove_custom_packager(packager=packager)
 
+    def create_api_gateway(
```

Comment:
Why not get an `APIGateway` object here instead of all of its components?

Reply:
@theSaarco I was thinking that it will be more intuitive for user to do 
`project.create_api_gateway(params..)`
 than
 ```
 gw = APIGateway(params)
 project.create_api_gateway(gw)
 ```

Reply:
@rokatyy - yeah, both ways make sense, and it's certainly ok to use this approach. Why not support both options? Allow passing either an `APIGateway` class or the set of params? 
There are use-cases where you already have an APIGateway object (for example - I got the existing GW from DB, modified it and now want to pass it with the changes).

Reply:
@theSaarco currently we don't have a gateway entity in DB, so I think we can keep it as simple as possible and gradually complicate the logic (in following PRs)

Reply:
But that's definitely a good idea